crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -363,131 +317,135 @@ fn check_class_attribute(checker: &mut Checker, attribute_expr: &ExprAttribute)
+fn check_context_key_usage_in_subscript(checker: &mut Checker, subscript: &ExprSubscript) {
+    if !in_airflow_task_function(checker.semantic()) {
         return;
     }
 
-    if attr.as_str() != "get" {
+    let ExprSubscript { value, slice, .. } = subscript;
+
+    let Some(ExprStringLiteral { value: key, .. }) = slice.as_string_literal_expr() else {
+        return;
+    };
+
+    let is_kwarg_parameter = value
+        .as_name_expr()
+        .is_some_and(|name| is_kwarg_parameter(checker.semantic(), name));
+
+    let is_assigned_from_get_current_context =
+        typing::resolve_assignment(value, checker.semantic()).is_some_and(|qualified_name| {
+            matches!(
+                qualified_name.segments(),
+                ["airflow", "utils", "context", "get_current_context"]
+            )
+        });
+
+    if !(is_kwarg_parameter || is_assigned_from_get_current_context) {
         return;
     }
 
-    for removed_key in REMOVED_CONTEXT_KEYS {
-        if let Some(argument) = call_expr.arguments.find_positional(0) {
-            if let Expr::StringLiteral(ExprStringLiteral { value, .. }) = argument {
-                if value == removed_key {
-                    checker.diagnostics.push(Diagnostic::new(
-                        Airflow3Removal {
-                            deprecated: removed_key.to_string(),
-                            replacement: Replacement::None,
-                        },
-                        argument.range(),
-                    ));
-                }
-            }
-        }
+    if REMOVED_CONTEXT_KEYS.contains(&key.to_str()) {
+        checker.diagnostics.push(Diagnostic::new(
+            Airflow3Removal {
+                deprecated: key.to_string(),
+                replacement: Replacement::None,
+            },
+            slice.range(),
+        ));
     }
 }
 
-/// Check whether the function is decorated by @task
-///
-///
-/// Examples for the above patterns:
-/// ```python
-/// from airflow.decorators import task
-///
-///
-/// @task
-/// def access_invalid_key_task_out_of_dag(**context):
-///     print("access invalid key", context.get("conf"))
-/// ```
-fn is_taskflow(checker: &mut Checker) -> bool {
-    let mut parents = checker.semantic().current_statements();
-    if let Some(Stmt::FunctionDef(StmtFunctionDef { decorator_list, .. })) =
-        parents.find(|stmt| stmt.is_function_def_stmt())
-    {
-        for decorator in decorator_list {
-            if checker
-                .semantic()
-                .resolve_qualified_name(map_callable(&decorator.expression))
-                .is_some_and(|qualified_name| {
-                    matches!(qualified_name.segments(), ["airflow", "decorators", "task"])
-                })
-            {
-                return true;
-            }
-        }
-    }
-    false
+/// Finds the parameter definition for a given name expression in a function.
+fn is_kwarg_parameter(semantic: &SemanticModel, name: &ExprName) -> bool {
```

Comment:
Just want to make sure that this is correct before merging - the update here is that the `**` parameter isn't limited to `context` and `kwargs`, so the following is valid as well:

```python
@task 
def some_task(**variables):
	variables.get('conf')
```

Is this correct? Or, do we need to limit it to `context` and `kwargs`?

Reply:
no, `variables` won't work. it should be limited to `context` and kwargs

Reply:
ah, let me try with `variables` again. It reminds me `kwargs` is a convention but not a keyword 🤦‍♂️

Reply:
Got it, thanks

Reply:
right, anything starts with `**` should be caught